### PR TITLE
Drop random SHA and update gonum.org/v1/gonum to v0.6.2

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This is part of the code organization effort to move away from random SHA(s) in our dependencies

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
/sig api-machinery
/priority important-soon

(used in pkg/controller/garbagecollector)

File: vendor/gonum.org/v1/gonum/lapack/gonum/dormr2.go [hunk 1]
```diff
@@ -8,10 +8,10 @@ import "gonum.org/v1/gonum/blas"
...
 // Dormr2 multiplies a general matrix C by an orthogonal matrix from a RQ factorization
 // determined by Dgerqf.
-//  C = Q * C    if side == blas.Left and trans == blas.NoTrans
-//  C = Q^T * C  if side == blas.Left and trans == blas.Trans
-//  C = C * Q    if side == blas.Right and trans == blas.NoTrans
-//  C = C * Q^T  if side == blas.Right and trans == blas.Trans
+//  C = Q * C   if side == blas.Left and trans == blas.NoTrans
+//  C = Qᵀ * C  if side == blas.Left and trans == blas.Trans
+//  C = C * Q   if side == blas.Right and trans == blas.NoTrans
+//  C = C * Qᵀ  if side == blas.Right and trans == blas.Trans
```
Comment (BenTheElder) [inline vendor/gonum.org/v1/gonum/lapack/gonum/dormr2.go]: cc @mattjmcnaughton @oomichi xref: https://github.com/kubernetes/kubernetes/pull/86987 :upside_down_face:

Comment (oomichi) [reply vendor/gonum.org/v1/gonum/lapack/gonum/dormr2.go]: Hehe, a nice point ;-)

**Comment by dims** [comment]
/assign @thockin @liggitt 

(for root approval)

**Pull request merged**